workbench-core/datasets/src/ddbDataSetMetadataPlugin.ts

```diff
@@ -122,7 +143,29 @@ export class DdbDataSetMetadataPlugin implements DataSetMetadataPlugin {
    * @param dataSetId - the ID of the Dataset to remove.
    */
   public async removeDataSet(dataSetId: string): Promise<void> {
-    await this._aws.helpers.ddb.delete(buildDynamoDBPkSk(dataSetId, this._dataSetKeyType)).execute();
+    const data = await this._aws.helpers.ddb.deleteItem({
+      key: buildDynamoDBPkSk(dataSetId, this._dataSetKeyType),
+      params: { return: 'ALL_OLD' }
+    });
+
+    const remainingDatasets = await this._storeStorageLocationToDdb(
+      {
+        name: data.storageName as string,
+        type: data.storageType as string,
+        awsAccountId: data.awsAccountId as string,
+        region: data.region as string
+      },
+      'decrement'
+    );
+
+    if (remainingDatasets < 1) {
+      await this._aws.helpers.ddb
+        .delete({
+          pk: this._dataSetKeyType,
```

Comment:
+1 ^ limit pk collision

Reply:
storage location isnt unique to a specific dataset, so we cant use the dataset id in our `pk`/`sk`. I want 100% sure what to label as `pk` and `sk`. I ended up with the approach of `pk` being the dataset key type to signify that its part of the datasets entries, and `sk` to separate the different storage locations.

Reply:
understand, this should still be prefixed with something so you can limit potential collisions

Reply:
Would something like `pk: '${this._dataSetKeyType}#${this._storageLocationKeyType}'` work?

Reply:
if storageLocationKeyType is accessible during query time or doesn't impact your query patterns then yeah that is fine